iDiff/utils/image_prep_utils.go

```diff
@@ -115,11 +117,18 @@ func getHistory(imgPath string) ([]string, error) {
 
 func getImageFromTar(tarPath string) (string, error) {
 	glog.Info("Extracting image tar to obtain image file system")
-	err := ExtractTar(tarPath)
-	if err != nil {
-		return "", err
-	}
 	path := strings.TrimSuffix(tarPath, filepath.Ext(tarPath))
+	args := []string{"iDiff/undocker.py", "--tar", tarPath, "-o", path, "-v"}
+	undockerCmd := exec.Command("python", args...)
```

Comment:
Ideally "undockering" the files will be done in native Go Code.  Understandably porting the code over costs time, I personally still think this is worthwhile as it appears there is not too much functional code that would need to be ported:
https://github.com/larsks/undocker/blob/master/undocker.py.   If absolutely necessary, the python file can be stored inside of the packaged go binary using go-bindata: https://github.com/jteeuwen/go-bindata.  "undocker" does not seem to require any additional packages so no additional setup should be required.  Python is standard in most Linux distros, this won't work out of the box though on Windows, I'm not sure about OSX. Also this python 2 specific code though, I don't think it will work with Python 3.  With this in mind, porting the code over would be ideal but we can definitely discuss this.  Also I don't think that the undocker.py file is currently in the PR.